core/src/wallet/ethereum/EthereumLikeAccount.cpp

```diff
@@ -459,23 +459,16 @@ namespace ledger {
                 auto log = logger();
 
                 log->debug(" Start erasing data of account : {}", getAccountUid());
+
+                std::lock_guard<std::mutex> lock(_synchronizationLock);
```

Comment:
This lock mechanism effort is duplicated for all currencies. It can be easy to make a mistake or forget about it.
Maybe the `AbstractAccount::eraseDataSince` could have that logic instead of moving it down to the coin? Even though `AbstractAccount::eraseDataSince` is virtual pure, you can still provide it an implementation that coins could call in the implementation?

Reply:
We agree but this needs moving the lock and event bus on abstract level, the impact could be huge, we should have a dedicated PR for this kind of changes (even should be part of segregation future tasks) !